doc/api_assets/api.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+    
+// Check if we have Javascript support
+document.querySelector(':root').classList.add('has-js');
+
+// Restore user mode preferences
+const kCustomPreference = 'customDarkTheme';
+const userSettings = sessionStorage.getItem(kCustomPreference);
+const themeToggleButton = document.getElementById('theme-toggle-btn');
+if (userSettings === null && window.matchMedia) {
+  const mq = window.matchMedia('(prefers-color-scheme: dark)');
+  if ('onchange' in mq) {
+    function mqChangeListener(e) {
+      document.documentElement.classList.toggle('dark-mode', e.matches);
+    }
+    mq.addEventListener('change', mqChangeListener);
+    if (themeToggleButton) {
+      themeToggleButton.addEventListener('click', function() {
+        mq.removeEventListener('change', mqChangeListener);
+      }, { once: true });
+    }
+  }
+  if (mq.matches) {
+    document.documentElement.classList.add('dark-mode');
+  }
+} else if (userSettings === 'true') {
+  document.documentElement.classList.add('dark-mode');
+}
+if (themeToggleButton) {
+  themeToggleButton.hidden = false;
+  themeToggleButton.addEventListener('click', function() {
+    sessionStorage.setItem(
+      kCustomPreference,
+      document.documentElement.classList.toggle('dark-mode')
+    );
+  });
+}
+
+// Handle pickers with click/taps rather than hovers
+const pickers = document.querySelectorAll('.picker-header');
+for(const picker of pickers) {
```

Comment:
```suggestion
for (const picker of pickers) {
```

Reply:
If we're introducing JavaScript in api_assets, I wonder if we want to enable ESLint for it.

Reply:
(I'm not sure how much modification would be required to our current default set of rules to accommodate web JS.)